test/session.test.js

```diff
@@ -345,3 +345,58 @@ test('should update the expires property of the session using Session#touch() ev
 
   t.true(sessionExpires1 !== sessionExpires2)
 })
+
+test('should use custom sessionId generator if available (with request)', async (t) => {
+  const fastify = Fastify()
+  fastify.register(fastifyCookie)
+  fastify.register(fastifySession, {
+    secret: 'cNaoPYAwF60HZJzkcNaoPYAwF60HZJzk',
+    cookie: { secure: false, maxAge: 10000 },
+    idGenerator: (request) => {
+      if (request?.session.returningVisitor) {
```

Comment:
Optional chaining is not supported by Node.js 10

Reply:
Fixed in
https://github.com/fastify/session/pull/64/commits/99614dad9a9cd78d6bc0daec75ba64d4723ebff9